include/swift/AST/Module.h

```diff
@@ -744,7 +748,7 @@ class FileUnit : public DeclContext {
   // Only allow allocation of FileUnits using the allocator in ASTContext
   // or by doing a placement new.
   void *operator new(size_t Bytes, ASTContext &C,
-                     unsigned Alignment = alignof(FileUnit));
+                     unsigned Alignment = alignof(FileUnit&));
```

Comment:
Hmm, something like this:
```
class FileUnit {
public:
  // Only allow allocation of FileUnits using the allocator in ASTContext
  // or by doing a placement new.
  void *operator new(size_t Bytes, ASTContext &C,
                     unsigned Alignment = alignOfFileUnit());
};

static inline unsigned alignOfFileUnit() {
	return alignof(FileUnit);
}
```

doesn't work due to same error:

>1>C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(757): error C2259: 'swift::FileUnit': cannot instantiate abstract class (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(757): note: due to following members: (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(757): note: 'void swift::FileUnit::lookupValue(swift::ModuleDecl::AccessPathTy,swift::DeclName,swift::NLKind,llvm::SmallVectorImpl<swift::ValueDecl *> &) const': is abstract (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(556): note: see declaration of 'swift::FileUnit::lookupValue' (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(757): note: 'void swift::FileUnit::lookupObjCMethods(swift::ObjCSelector,llvm::SmallVectorImpl<swift::AbstractFunctionDecl *> &) const': is abstract (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(588): note: see declaration of 'swift::FileUnit::lookupObjCMethods' (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(757): note: 'swift::Identifier swift::FileUnit::getDiscriminatorForPrivateValue(const swift::ValueDecl *) const': is abstract (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)
1>  C:\Users\Lukey\Documents\GitHub\my-swift\swift\include\swift/AST/Module.h(631): note: see declaration of 'swift::FileUnit::getDiscriminatorForPrivateValue' (compiling source file C:\Users\Lukey\Documents\GitHub\my-swift\swift\lib\AST\ASTPrinter.cpp)